js/data/buffer.js

```diff
@@ -170,18 +169,16 @@ Buffer.prototype._createPushMethod = function() {
         body += '\nvar ' + offsetId + ' = (offset + ' + attribute.offset + ') / ' + attribute.type.size + ';\n';
 
         for (var j = 0; j < attribute.components; j++) {
-            var valueId = 'value' + i + '_' + j;
+            var rvalue = 'value.' + attribute.name + '[' + j + ']';
             var lvalue = 'this.views.' + attribute.type.name + '[' + offsetId + ' + ' + j + ']';
 
-            body += lvalue + ' = ' + valueId + ';\n';
-
-            argNames.push(valueId);
+            body += lvalue + ' = ' + rvalue + ';\n';
         }
     }
 
     body += '\nreturn index;\n';
 
-    this.push = new Function(argNames, body);
+    this.push = new Function('value', body);
```

Comment:
That's a step in the right direction. I think `push('attrName', a, b, c)` is a problematic API: when do we advance to the next item? Maybe we should move to a `set(index, attrName, a, b, c)` paradigm.